db/dbScript.sql

```diff
@@ -0,0 +1,393 @@
+	IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'not_type') THEN
+        CREATE TYPE not_type AS ENUM ('message', 'alert', 'reminder');
+    END IF;
+END $$;
+
+--Since we will have a lot of gyms stored (potentially) it does not make sense to store working hours for each of them as a text
+--Especially since there are significantly smaller potential patterns [(opening hours, closing hours) pairs] in comparison to the potential number of gyms stored
+CREATE TABLE if not exists Working_hours
+(
+  id uuid NOT null DEFAULT uuid_generate_v4(),--uuid is analogous to id, but uuid is pseudo-random and the length and format of uuids are the same
+  open_from TIME NOT NULL,
+  open_until TIME NOT NULL,
+  PRIMARY KEY (id),
+  UNIQUE (open_from),
+  UNIQUE (open_until),
+  check (open_until > open_from)
+);
+
+CREATE TABLE if not exists Currency
+(
+  id uuid NOT null DEFAULT uuid_generate_v4(),
+  name VARCHAR(10) NOT NULL,
+  code CHAR(3) NOT NULL,
+  PRIMARY KEY (id),
+  UNIQUE (name),
+  UNIQUE (code)
+);
+
+CREATE TABLE if not exists Account
+(
+  id uuid NOT null DEFAULT uuid_generate_v4(),
+  username VARCHAR(40) NOT null check (length(username) > 5),
+  email VARCHAR(60) not NULL,
+  first_name VARCHAR(60) NOT NULL,
+  last_name VARCHAR(60) NOT NULL,
+  created_at TIMESTAMP NOT null check (created_at <= NOW()),
+  provider provider_type NOT NULL, --we want to differentiate between google accounts and local accounts
+  password_hash CHAR(60) NOT NULL, --we are storing hash of the password that we got from bycrypt
+  type account_type NOT NULL, --we want to differentate between user, admin and gym accounts
+  created_by uuid, --admin accounts can be created only by another admin accounts. in this case we want to register who created the admin account
+  PRIMARY KEY (id),
+  FOREIGN KEY (created_by) REFERENCES Account(id),
+  UNIQUE (username),
+  UNIQUE (email)
+);
+create index if not exists idx_account_username on Account(username);
+create index if not exists idx_account_email on Account(email);
+create index if not exists idx_account_type on Account("type");
+
+create table if not exists Country
+(
+	id uuid not null default uuid_generate_v4(),
+	name varchar(56) not null,
+	primary key(id),
+	UNIQUE (name)
+);
+create index if not exists idx_country_name on Country("name");
+
+create table if not exists City
+(
+	id uuid not null default uuid_generate_v4(),
+	nelatitude FLOAT NOT null check (nelatitude between -90 and 90), --rectangular approximation of the area of the city
+    nelongitude FLOAT NOT null check (nelongitude between -180 and 180),
+    swlatitude FLOAT NOT null check (swlatitude between -90 and 90),
+    swlongitude FLOAT NOT null check (swlongitude between -180 and 180),
+	name varchar(100) not null,
+	country_id uuid not null,
+	primary key(id),
+	foreign key (country_id) references Country(id)
+);
+create index if not exists idx_city_name_country_id on City("name", country_id);
+
+CREATE TABLE if not exists Gym
```

Comment:
We might consider adding a description of the available gym equipment, potentially in a separate table. This could include details such as the number of barbells, treadmills, or availability of a swiming pool, etc.

Reply:
We would have the same problem as with gym prices (as it turns out). Since we retrieve all the gyms in the city dynamically, and Google Maps do not provide prices, gym equipment or anything else, we would need to somehow obtain this information either manually or through Web Scraping or through some third-party services. 

If we do not aim to provide this information automatically, we can add this option for the gym accounts to fill in manually. But personally I would limit it to the text field, at least for now (it is additional work, and we still have a lot of things to do)